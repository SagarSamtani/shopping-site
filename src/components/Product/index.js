import React from "react";
import Card from "@material-ui/core/Card";
import productStyles from "styles/product";
import Button from "@material-ui/core/Button";
import CardMedia from "@material-ui/core/CardMedia";
import { makeStyles } from "@material-ui/core/styles";
import CardHeader from "@material-ui/core/CardHeader";
import Typography from "@material-ui/core/Typography";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";

const useStyles = makeStyles(theme => productStyles(theme));

const Product = props => {
  const { productInfo, onClickHandler } = props,
    { name, image, price, currency, description } = productInfo;

  const classes = useStyles();

  return (
    <Card className={classes.card}>
      <CardHeader className={classes.cardHeader} title={name} />
      <CardMedia
        className={classes.media}
        image={require(`assets/images/${image}`)}
        alt={image}
        title={image}
      />
      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {description}
        </Typography>
      </CardContent>
      <CardActions className={classes.actionItems}>
        <Typography className={classes.price} component="span">
          {currency}{price}
        </Typography>
        <Button
          onClick={onClickHandler}
          variant="outlined"
          className={classes.addBtn}>
          Add to Cart
        </Button>
      </CardActions>
    </Card>
  );
};

export default Product;
