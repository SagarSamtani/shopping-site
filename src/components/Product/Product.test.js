import React from "react";
import renderer from "react-test-renderer";
import Product from "./index";

const mockProductData = {
  id: 2,
  name: "Wallet",
  image: "wallet.png",
  price: "299",
  currency: "â‚¹",
  description: "It is Woodland wallet"
};

const mockCartFunc = paramProduct => {
  console.log("paramProduct", paramProduct);
};

describe("Counter", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <Product
        key={mockProductData.id}
        productInfo={mockProductData}
        onClickHandler={() => {
          mockCartFunc(mockProductData);
        }}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
