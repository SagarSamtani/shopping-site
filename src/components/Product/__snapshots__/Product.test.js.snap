// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root makeStyles-cardHeader-3"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Wallet
      </span>
    </div>
  </div>
  <div
    alt="wallet.png"
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"wallet.png\\")",
      }
    }
    title="wallet.png"
  />
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      It is Woodland wallet
    </p>
  </div>
  <div
    className="MuiCardActions-root makeStyles-actionItems-6 MuiCardActions-spacing"
  >
    <span
      className="MuiTypography-root makeStyles-price-4 MuiTypography-body1"
    >
      â‚¹
      299
    </span>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-addBtn-5"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Add to Cart
      </span>
    </button>
  </div>
</div>
`;
